packages/@okta/vuepress-site/docs/reference/api/roles/index.md

```diff
@@ -1362,5 +2624,177 @@ A Role could either be assigned to the User directly or be assigned to a Group o
 
 | Assignment type     | Description                                                        |
 | :------------------ | :----------------------------------------------------------------- |
-| `GROUP`             | Role was assigned to an admin Group of which the User is a member |
+| `GROUP`             | Role was assigned to an admin Group of which the User is a member  |
 | `USER`              | Role was assigned to the User directly                             |
+
+## Custom role object
+<ApiLifecycle access="beta" />
+A custom role is a custom set of [permissions](#permission-types). A custom role is uniquely identified within your org by its id or label.
+
+### Custom role properties
+| Property         | Description                         | DataType                                       | Nullable   | Unique   | Read Only |
+| :--------------- | :---------------------------------- | :--------------------------------------------- | :--------- | :------- | :-------- |
+| id               | Unique key for the role             | String                                         | FALSE      | TRUE     | TRUE      |
+| label            | Display name of Role                | String                                         | FALSE      | TRUE     | FALSE     |
+| permissions      | the permissions the new role grants | Array of [Permission types](#permission-types) | FALSE      | FALSE    | FALSE     |
+
+#### Example
+
+```json
+{
+    "id": "cr0Yq6IJxGIr0ouum0g3",
+    "label": "UserCreator",
+    "description": "Create users",
+    "permissions": [
+      "okta.users.create",
+      "okta.users.read",
+      "okta.groups.read",
+      "okta.users.profile.manage"
+    ]
+}
+```
+### Permission types
+<ApiLifecycle access="beta" />
+
+Permissions can be used to build custom roles. Permissions to manage a resource also grant the viewing privileges for the the same resource so you won't need to assign them separately.
+
+User permissions are all only effective with respect to the group(s) to which the admin is granted role via the resource set assignments.
+
+| Assignment type                     | Description                                                                                                                         |
```

Comment:
worth a column for "Applies to Resources of Type" ?

Reply:
added